torchgeo/transforms/temporal.py

```diff
@@ -68,3 +72,42 @@ def compute_transformation(
         """
         out = self.identity_matrix(input)
         return out
+
+
+class CyclicalEncoder(nn.Module):
+    """Generic sinusoidal embedding for periodic temporal features."""
+
+    def __init__(
+        self,
+        period: pd.Timedelta,
+        time_key: str = 'time',
+        out_key: str = 'time_embedding',
```

Comment:
Let's split the output into two keys. May not even need an `out_key` parameter. Could just use `f"sin_{time_key}"` and `f"cos_{time_key}"`.